File: java/src/main/java/com/fengtian/qianer/atcoder/abc/abc163/ABC163_B.java
package com.fengtian.qianer.atcoder.abc.abc163;

import java.util.Arrays;
import java.util.Scanner;

public class ABC163_B {

    public static void main(String[] args) {
        final Scanner sc = new Scanner(System.in);

        final int n = sc.nextInt();
        final int m = sc.nextInt();

        final int[] a = new int[m];
        for (int i = 0; i < m; i++) {
            a[i] = sc.nextInt();
        }

        int sum = Arrays.stream(a).sum();
        System.out.println(sum > n ? -1 : n - sum);
    }
}

File: kotlin/src/abc/abc163/ABC163_B.kt
package abc.abc163

import java.util.*

fun main(args: Array<String>) {
    val sc = Scanner(System.`in`)

    val n = sc.nextInt()
    val m = sc.nextInt()

    val a = mutableListOf<Int>()
    (0 until m).forEach {
        a.add(sc.nextInt())
    }

    if (n >= a.sum()) {
        println(n - a.sum())
    } else {
        println(-1)
    }
}